dot/rpc/subscription/listeners.go

```diff
@@ -309,22 +311,28 @@ type ExtrinsicSubmitListener struct {
 	importedChan  chan *types.Block
 	importedHash  common.Hash
 	finalisedChan chan *types.FinalisationInfo
+	// txStatusChan is used to know when transaction/extrinsic becomes part of the
+	// ready queue or future queue.
+	// we are using transaction.PriorityQueue for ready queue and transaction.Pool
+	// for future queue.
+	txStatusChan  chan transaction.StatusNotification
 	done          chan struct{}
 	cancel        chan struct{}
 	cancelTimeout time.Duration
 }
 
 // NewExtrinsicSubmitListener constructor to build new ExtrinsicSubmitListener
-func NewExtrinsicSubmitListener(conn *WSConn, extBytes []byte) *ExtrinsicSubmitListener {
-	esl := &ExtrinsicSubmitListener{
+func NewExtrinsicSubmitListener(conn *WSConn, extBytes []byte, importedChan chan *types.Block, txStatusChan chan transaction.StatusNotification, finalisedChan chan *types.FinalisationInfo) *ExtrinsicSubmitListener {
+	return &ExtrinsicSubmitListener{
 		wsconn:        conn,
 		extrinsic:     types.Extrinsic(extBytes),
```

Comment:
it seems that `extBytes` could be multiple extrinsics, not just 1 (ie. the comment in `initExtrinsicWatch`). is this still the case? if so, you'll need to scale.Unmarshal the `extBytes` into `[]types.Extrinsic`